src/app/src/components/RegisterForm.jsx

```diff
@@ -173,63 +176,36 @@ function mapStateToProps({
     };
 }
 
-function mapDispatchToProps(dispatch) {
+const getStateFromEventForEventType = Object.freeze({
+    [inputTypesEnum.checkbox]: getCheckedFromEvent,
+    [inputTypesEnum.select]: identity,
+    [inputTypesEnum.text]: getValueFromEvent,
+    [inputTypesEnum.password]: getValueFromEvent,
+});
+
+const mapDispatchToProps = memoize((dispatch) => {
+    const makeInputChangeHandler = (field, getStateFromEvent) => e =>
+        dispatch(updateSignUpFormInput({
+            value: getStateFromEvent(e),
+            field,
+        }));
+
+    const inputUpdates = Object
+        .values(registrationFieldsEnum)
```

Comment:
Nice reuse of our existing fields list. We are creeping toward a forms library :-)